vector/src/main/java/im/vector/util/BugReporter.java

```diff
@@ -79,262 +76,227 @@
-    private static void sendBugReportWithMail(Context context, boolean withScreenshot, String bugDescription) {
-        Bitmap screenShot = takeScreenshot();
+                    try {
+                        // Read the SERVER RESPONSE
+                        mResponseCode = conn.getResponseCode();
+                    } catch (EOFException eofEx) {
+                        mResponseCode = HttpURLConnection.HTTP_INTERNAL_ERROR;
+                    }
 
-        if (null != screenShot) {
-            try {
-                String message = buildBugReportMessage(context, bugDescription);
-                File file = buildBugZipFile(context, withScreenshot, bugDescription);
-
-                // list the intent which supports email
-                // it should avoid having lot of unexpected applications (like bluetooth...)
-                Intent emailIntent = new Intent(Intent.ACTION_SENDTO, Uri.fromParts("mailto", "rageshake@riot.im", null));
-                emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Mail subject");
-                List<ResolveInfo> resolveInfos = context.getPackageManager().queryIntentActivities(emailIntent, 0);
-
-                if ((null == resolveInfos) || (0 == resolveInfos.size())) {
-                    Log.e(LOG_TAG, "Cannot send bug report because there is no application to send emails");
-                    return;
+                    // if the upload failed, try to retrieve the reason
+                    if (mResponseCode != HttpURLConnection.HTTP_OK) {
+                        serverError = null;
+                        InputStream is = conn.getErrorStream();
+
+                        if (null != is) {
+                            int ch;
+                            StringBuilder b = new StringBuilder();
+                            while ((ch = is.read()) != -1) {
+                                b.append((char) ch);
+                            }
+                            serverError = b.toString();
+                            is.close();
+
+                            // check if the error message
+                            try {
+                                JSONObject responseJSON = new JSONObject(serverError);
+                                serverError = responseJSON.getString("error");
+                            } catch (JSONException e) {
+                                Log.e(LOG_TAG, "doInBackground ; Json conversion failed " + e.getMessage());
+                            }
+
+                            // should never happen
+                            if (null == serverError) {
+                                serverError = "Failed with error " + mResponseCode;
+                            }
+                        }
+                    }
+                } catch (Exception e) {
+                    Log.e(LOG_TAG, "doInBackground ; failed with error " + e.getClass() + " - " + e.getMessage());
+                    serverError = e.getLocalizedMessage();
                 }
```

Comment:
Add a `finally `block to disconnect the `HttpURLConnection` after closing the inputstream?